### DISCUSSION TITLE
Slow charging, needs to plug charger before boot to be able to fast charge

### DISCUSSION THREAD

**Discussion by wegzi** [jusqua/galaxy-book4-linux#2] [General]
I found a possible fix for that issue. I haven't experienced this behavior myself, but you can try to fix it using this topic.

https://bbs.archlinux.org/viewtopic.php?pid=2183365#p2183365

You can also find some infos here
https://wiki.archlinux.org/title/Laptop/Samsung

**Comment by jusqua** [comment]
After some tinkering, this is a valid workaround.
I had some issues because I dual booted Windows without disabling fast-start, so I just turned off fast-start and everything was fine.

**Comment by Justbob9178** [comment]
If anyone else is having this issue and isn't dual booting windows adding `acpi_osi=\"Windows 2020\" acpi_enforce_resources=lax usbcore.autosuspend=-1` to your boot parameters (adding to the end of `GRUB_CMDLINE_LINUX_DEFAULT=` line in `/etc/default/grub` then `sudo update-grub`)will let you charge at ~7-30w instead of the 0-5w that is normal. This is by no means a solution but hopefully can help someone.


edit: os is debian 12 on samsung galaxy book 4 ([NP750XGJ-KG1UK](https://www.samsung.com/uk/computers/galaxy-book/galaxy-book4-15-6-inch-13th-i3-8gb-256gb-np750xgj-kg1uk/))

**Comment by Gugu7264** [comment]
I had the same issue with a galaxy book 4 on nixos, and the above solutions unfortunately didn't work for me.
However, also blacklisting some kernel modules worked !
```
boot.blacklistedKernelModules = [ "ucsi_acpi" "typec_ucsi" ];
```

See original topic [here](https://discourse.nixos.org/t/charging-power-limited-with-usb-c-only-when-powered-on/60193/10)

**Reply by Velasko** [reply]
Same for me, on Fedora Kinoite.
The equivalent on for the kernel argument was
`modprobe.blacklist=ucsi_acpi,typec_ucsi`
